StRoot/StFwdTrackMaker/StFwdTrackMaker.cxx

```diff
@@ -857,101 +1011,407 @@ int StFwdTrackMaker::Make() {
         mHistograms[ "nMcTracksFwdNoThreshold" ]->Fill( nForwardTracksNoThreshold );
     }
 
+
+    return nForwardTracks;
+} // loadMcTracks
+
+void StFwdTrackMaker::loadFcs( ) {
+    StEvent *stEvent = static_cast<StEvent *>(GetInputDS("StEvent"));
+    StFcsDb* fcsDb=static_cast<StFcsDb*>(GetDataSet("fcsDb"));
+    if ( !stEvent || !fcsDb ){
+        return;
+    }
+    StFcsCollection* fcsCol = stEvent->fcsCollection();
+    StEpdGeom epdgeo;
+
+
+    // LOAD ECAL / HCAL CLUSTERS
+    for ( int idet = 0; idet  < 4; idet++ ){
+        StSPtrVecFcsCluster& clusters = fcsCol->clusters(idet);
+        int nc=fcsCol->numberOfClusters(idet);
+        for ( int i = 0; i < nc; i++ ){
+            StFcsCluster* clu = clusters[i];
+            StThreeVectorD xyz = fcsDb->getStarXYZfromColumnRow(clu->detectorId(),clu->x(),clu->y());
+            mFcsClusters.push_back( TVector3( xyz.x(), xyz.y(), xyz.z() - 2 ) );
+            mFcsClusterEnergy.push_back( clu->energy() );
+        }
+    }
+
+    // LOAD PRESHOWER HITS (EPD)
+    for ( int det = 4; det < 6; det ++ ) {
+
+        StSPtrVecFcsHit& hits = stEvent->fcsCollection()->hits(det);
+        int nh=fcsCol->numberOfHits(det);
+        for ( int i = 0; i < nh; i++ ){
+            StFcsHit* hit=hits[i];
+
+            if(det==kFcsPresNorthDetId || det==kFcsPresSouthDetId){ //EPD
+                 double zepd=375.0;
+                 int pp,tt,n;
+                 double x[5],y[5];
+
+                 if ( hit->energy() < 0.2 ) continue;
+                 fcsDb->getEPDfromId(det,hit->id(),pp,tt);
+                 epdgeo.GetCorners(100*pp+tt,&n,x,y);
+                 double x0 = (x[0] + x[1] + x[2] + x[3]) / 4.0;
+                 double y0 = (y[0] + y[1] + y[2] + y[3]) / 4.0;
+                 mFcsPreHits.push_back( TVector3( x0, y0, zepd ) );
+            }
+        }
+    }
+} // loadFcs
+
+
+//________________________________________________________________________
+int StFwdTrackMaker::Make() {
+    // START time for measuring tracking
+    long long itStart = FwdTrackerUtils::nowNanoSecond();
+
+    // Access forward Tracker maps
+    FwdDataSource::McTrackMap_t &mcTrackMap = mForwardData->getMcTracks();
+    FwdDataSource::HitMap_t &hitMap = mForwardData->getFttHits();
+    FwdDataSource::HitMap_t &fsiHitMap = mForwardData->getFstHits();
+    
+    // clear vectors for visualization OBJ hits
+    mFttHits.clear();
+    mFstHits.clear();
+    mFcsPreHits.clear();
+    mFcsClusters.clear();
+    mFwdTracks.clear();
+
+    // default event vertex
+    mForwardTracker->setEventVertex( TVector3( 0, 0, 0 ) );
+
+    /**********************************************************************/
+    // Load MC tracks
+    size_t nForwardTracks = loadMcTracks( mcTrackMap );
```

Comment:
Is it also called when processing real data? Kind of strange if you load MC tracks, that's why I am asking

Reply:
this is safe to call, as it will do nothing if there are no MC Tracks/info. Is there a cleaner way to check if we are running with sim or embedding? If so I'll update, but this is safe as is.